dandiapi/api/views/asset.py

```diff
@@ -194,20 +199,26 @@ def update(self, request, versions__dandiset__pk, versions__version, **kwargs):
                 status=status.HTTP_405_METHOD_NOT_ALLOWED,
             )
 
-        serializer = AssetRequestSerializer(data=request.data)
+        serializer = AssetUpdateRequestSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
 
-        asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
-
-        metadata = serializer.validated_data['metadata']
-        if 'path' not in metadata:
-            return Response('No path specified in metadata', status=404)
-        path = metadata['path']
-        # Strip away any computed fields
-        metadata = Asset.strip_metadata(metadata)
-        asset_metadata, created = AssetMetadata.objects.get_or_create(metadata=metadata)
-        if created:
-            asset_metadata.save()
+        if 'blob_id' in serializer.validated_data:
+            asset_blob = get_object_or_404(AssetBlob, blob_id=serializer.validated_data['blob_id'])
+        else:
+            asset_blob = old_asset.blob
+
+        if 'metadata' in serializer.validated_data:
+            metadata = serializer.validated_data['metadata']
+            if 'path' not in metadata:
+                return Response('No path specified in metadata', status=404)
+            path = metadata['path']
+            # Strip away any computed fields
+            metadata = Asset.strip_metadata(metadata)
+            asset_metadata, created = AssetMetadata.objects.get_or_create(metadata=metadata)
+            if created:
+                asset_metadata.save()
+        else:
+            asset_metadata = old_asset.metadata
```

Comment:
I think we should just disallow PUT without metadata (in light of #392 and the fact that metadata might be completely wrong for the blob including the digests etc)

Reply:
If you open a new issue to address [the other thread](https://github.com/dandi/dandi-api/pull/382/files#r664671854) we can discuss this as part of that issue.